src/platform/forms/save-in-progress/actions.js

```diff
@@ -343,6 +345,8 @@ export function removeInProgressForm(formId, migrations, prefillTransformer) {
         recordEvent({
           event: `${trackingPrefix}sip-form-start-over`
         });
+        // This action removes the form from the profile list
+        dispatch({ type: REMOVING_SAVED_FORM_SUCCESS, formId });
```

Comment:
This is what makes it work.

It's needed now because of the re-mounting of `RoutedSavableApp`.